src/plugins/integrations/AWSIntegration.ts

```diff
@@ -276,9 +280,12 @@ class AWSIntegration implements Integration {
               activeSpan.setTag(SpanTags.TOPOLOGY_VERTEX, true);
               activeSpan.setTag(SpanTags.TRIGGER_DOMAIN_NAME, LAMBDA_APPLICATION_DOMAIN_NAME);
               activeSpan.setTag(SpanTags.TRIGGER_CLASS_NAME, LAMBDA_APPLICATION_CLASS_NAME);
+
+              activeSpan.setTag(AwsKinesisTags.STREAM_NAME, streamName);
             }
           } else if (serviceName === 'firehose') {
-            const streamName = request.params ? request.params.DeliveryStreamName : AWS_SERVICE_REQUEST;
+            const streamName = koalas(request.params.DeliveryStreamName, AWS_SERVICE_REQUEST);
```

Comment:
If stream name is not valid for the request, it is ok to use `AWS_SERVICE_REQUEST` as the default span name, but it should not be used for `AwsFirehoseTags. STREAM_NAME` tag. So you may handle this by introducing another local variable for span name

Reply:
Done. Introduced a `spanName` variable.